packages/ir-sdk/fern/apis/ir-types-latest/definition/ir.yml

```diff
@@ -77,11 +77,24 @@ types:
       hasPaginatedEndpoints: boolean
       hasFileDownloadEndpoints: boolean
       platformHeaders: PlatformHeaders
+      version: optional<VersionDeclaration>
   PlatformHeaders:
     properties:
       language: string
       sdkName: string
       sdkVersion: string
+  VersionDeclaration:
+    docs: |
+      The available set of versions for the API. This is used to generate a special
+      enum that can be used to specify the version of the API to use. This information
+      is sent as an HTTP header (e.g. X-API-Version) on every request.
+
+      If the enum does _not_ define a default value, the version should be treated like
+      a required global header parameter. The version header should also support any
+      environment variable scanning specified by the header.
+    properties:
+      header: http.HttpHeader
+      value: types.EnumTypeDeclaration
```

Comment:
I would type this as a union to make things easier for the future: 

```
ApiVersionScheme: 
  union: 
     headerBased: HeaderBasedAPIVersioningScheme
     
HeaderBasedAPIVersioningScheme: 
  properties: 
    header: http.HttpHeader
```

Reply:
Sweet, I had this beforehand but thought it might be overkill because we can always migrate to it. But given that you suggested the exact same, I'm happy to go back to this.